### PR TITLE
Grid lines

### PR DESCRIPTION
**Pull request by jotoh98**
- added grid lines
- canvas boundary
- new geometry style defaults api: register own styles
fixes #156

### CODE REVIEW

**Comment by 1-alex98** [comment]
fix failing CI

**Comment by jotoh98** [comment]
ci fixed

File: src/main/java/com/treasure/hunt/jts/geom/Grid.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    /**
+     * Minimal rendering distance between two grid lines.
+     */
+    private double minDistance = 40d;
+
+    /**
+     * Maximal rendering distance between two grid lines.
+     */
+    private double maxDistance = 80d;
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/jts/geom/Grid.java]: constants

File: src/main/java/com/treasure/hunt/jts/geom/Grid.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    /**
+     * Font for grid line identification number.
+     */
+    private Font font = new Font(Font.SANS_SERIF, Font.PLAIN, 11);
+
+    /**
+     * Formatter for grid line identification numbers.
+     */
+    private DecimalFormat df = new DecimalFormat("###.###");
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/jts/geom/Grid.java]: rename

File: src/main/java/com/treasure/hunt/utils/RenderUtils.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import java.awt.*;
+import java.awt.font.GlyphVector;
+import java.awt.geom.AffineTransform;
+import java.awt.image.BufferedImage;
+
+@Slf4j
+public class RenderUtils {
+
+    public static Shape shapeFromText(Font font, String string, double x, double y) {
+        AffineTransform tx = new AffineTransform();
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/utils/RenderUtils.java]: rename

File: src/main/java/com/treasure/hunt/utils/RenderUtils.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        tx.translate(x, y);
+        return tx.createTransformedShape(shapeFromText(font, string));
+    }
+
+    public static Shape shapeFromText(Font font, String string) {
+        BufferedImage img = new BufferedImage(100, 100, BufferedImage.TYPE_INT_ARGB);
+        Graphics2D g2 = img.createGraphics();
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/utils/RenderUtils.java]: rename

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Review by 1-alex98 — APPROVED** [review]

**Pull request merged**